internal/store/store.go

```diff
@@ -509,6 +510,59 @@ func (s *Store) PushCompose(composeID uuid.UUID, bp *blueprint.Blueprint, checks
 	return nil
 }
 
+// PushComposeRequest is an alternative to PushCompose which does not assume a pre-defined distro, as such it is better
+// suited for RCM API and possible future API that would respect the fact that we can build any distro and any arch
+func (s *Store) PushComposeRequest(request common.ComposeRequest) error {
+	// Create a local target that will be appended to the requested targets
+	appendTargets := []*target.Target{}
+	if s.stateDir != nil {
+		appendTargets = append(appendTargets, target.NewLocalTarget(
+			&target.LocalTargetOptions{
+				Location: *s.stateDir + "/outputs/" + request.ComposeID.String(),
+			},
+		))
+	}
+
+	// This should never happen and once distro.Pipeline is refactored this check will go away
+	arch, exists := request.Arch.ToString()
+	if !exists {
+		panic("fatal error, arch should exist but it does not")
+	}
+
+	for range request.RequestedImages {
+		// TODO: handle custom upload targets
+		//targets := append(imgRequest.UpTarget, appendTargets)
+		// TODO: implment support for no checksums
+		// TODO: implement compatibility translation from image type or refactor distro packages
+		pipeline, err := s.distro.Pipeline(&request.Blueprint, request.Repositories, nil, arch, "qcow2", 0)
+		if err != nil {
+			return err
+		}
+		// TODO: refactor compose so that it can contain multiple image builds with different types and upload targets
+		// TODO: per image build
```

Comment:
Overall, I like this very much. It is ok to leave missing functionality out as TODO, but in this case I think a couple of these TODOs must be implemented in earlier commits, because as it stands this would lead to invalid behaviour. Firstly, I think you need to resolve the checksum (as mentioned in previous comments) and pass them in. Having a `nil` checksum would lead to an invalid `pipeline`. Secondly, I think you need to either refuse composes with more than one request, or implement multiple job support in a previous commit. You are right that these things are now indexed by `ComposeID` and that's not enough by itself. I think we should introduce a separate JobID, which is just a random UUID in the same way that `ComposeID` is (so from the workers this makes no difference), and then have a mapping back from `JobID`s to `ComposeID`s internally.
In order to get the rest of this merged, I think it would be ok to leave that as a TODO and instead just error out in case we are passed more than one request at once.

Reply:
I introduced a panic in the beginning of this function to prevent it from being used. I think we should not implement support for no checksum as part of this PR because it is huge already and the cost of rebasing it is far greater then the benefits of not merging it now. The RCM API is not yet ready anyway and this decision is not going to affect this fact. If we merge this without support for no checksums it will enable us to focus on implementing it instead of keeping the changes synchronized with master branch.